state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1363,222 +1474,200 @@ pub mod pallet {
+					input_amount.into(),
 					params.min_price,
 				))
 				.unwrap_or(u128::MAX),
 			});
 
-			let (swap_id, execute_at) = Self::schedule_swap(
-				from,
-				to,
-				net_amount,
-				refund_params,
-				SwapType::Swap(destination_address.clone()),
-			);
-
-			for Beneficiary { account, bps } in broker_commission {
-				EarnedBrokerFees::<T>::mutate(&account, from, |earned_fees| {
-					earned_fees.saturating_accrue(
-						Permill::from_parts(bps as u32 * BASIS_POINTS_PER_MILLION) * amount,
-					)
-				});
-			}
-
-			Self::deposit_event(Event::<T>::SwapScheduled {
-				swap_id,
-				source_asset: from,
-				deposit_amount: amount,
-				destination_asset: to,
-				destination_address: encoded_destination_address,
-				origin: swap_origin,
-				swap_type: SwapType::Swap(destination_address),
-				broker_commission: Some(fee),
-				broker_fee: Some(fee),
-				execute_at,
-			});
-
-			swap_id
-		}
-	}
+			let request_type_for_event = match request_type {
+				SwapRequestType::NetworkFee => {
+					let _swap_id = Self::schedule_swap(
+						input_asset,
+						output_asset,
+						net_amount,
+						None,
+						SwapType::NetworkFee,
+						request_id,
+					);
 
-	impl<T: Config> CcmHandler for Pallet<T> {
-		fn on_ccm_deposit(
-			source_asset: Asset,
-			deposit_amount: AssetAmount,
-			destination_asset: Asset,
-			destination_address: ForeignChainAddress,
-			deposit_metadata: CcmDepositMetadata,
-			origin: SwapOrigin,
-			// TODO: CCM should use refund params
-			_refund_params: Option<ChannelRefundParameters>,
-		) -> Result<CcmSwapIds, ()> {
-			let encoded_destination_address =
-				T::AddressConverter::to_encoded_address(destination_address.clone());
-			// Caller should ensure that assets and addresses are compatible.
-			debug_assert!(destination_address.chain() == ForeignChain::from(destination_asset));
-
-			let CcmSwapAmounts { principal_swap_amount, gas_budget, other_gas_asset } =
-				match Self::principal_and_gas_amounts(
-					deposit_amount,
-					&deposit_metadata.channel_metadata,
-					source_asset,
-					destination_asset,
-				) {
-					Ok(amounts) => amounts,
-					Err(reason) => {
-						// Confiscate the deposit and emit an event.
-						CollectedRejectedFunds::<T>::mutate(source_asset, |fund| {
-							*fund = fund.saturating_add(deposit_amount)
-						});
-
-						Self::deposit_event(Event::<T>::CcmFailed {
-							reason,
-							destination_address: encoded_destination_address,
-							deposit_metadata,
-							origin: origin.clone(),
-						});
-						return Err(())
-					},
-				};
+					SwapRequests::<T>::insert(
+						request_id,
+						SwapRequest {
+							id: request_id,
+							input_asset,
+							output_asset,
+							refund_params: None,
+							state: SwapRequestState::NetworkFee,
+						},
+					);
 
-			let ccm_id = CcmIdCounter::<T>::mutate(|id| {
-				id.saturating_accrue(1);
-				*id
-			});
+					RequestTypeEncoded::NetworkFee
+				},
+				SwapRequestType::IngressEgressFee => {
```

Comment:
It feels like there's a lot of repetition in these branches. Could we not just define schedule_swap as `schedule_swap(request_id, &swap_request)` and then we can avoid having this huge match statement?.

Ideally though, schedule_swap should not have to care about the originating swap request. We should probably add a `fees: Vec<Fee>` member that is stored with every request. Then that can be propagated when we schedule the swap, and the request *type* doesn't matter.

This will make it easier to take the broker fees in USDC, as is the plan. (we just add a `Fee::BrokerFee {id,rate}`).

Reply:
> Could we not just define schedule_swap as schedule_swap(request_id, &swap_request)

not sure how to reconcile this with 

` schedule_swap should not have to care about the originating swap request`

maybe we could discuss this in a synchronous conversation when you are back.

> We should probably add a fees: Vec<Fee> member that is stored with every request.

Yeah, that sounds like a good idea, I will give it a go.